src/app/data-model/Encode.h

```diff
@@ -90,6 +92,44 @@ CHIP_ERROR Encode(TLV::TLVWriter & writer, TLV::Tag tag, const X & x)
     return x.Encode(writer, tag);
```

Comment:
Do we want to add a !FabricScoped check to the SFINAE here, or just assume it will fail by virtue of no Encode method?

Reply:
I did do this originally, but quickly realized that `DataModel::IsFabricScoped<X>::value` isn't actually available for non-fabric-scoped complex types (like commands for example).

Either we could emit `static constexpr bool kIsFabricScoped = false;` universally for all emitted cluster objects, OR, let the compiler just fail naturally anyways given the absence of the Encode method.

I'd be fine either way honestly...

Reply:
The former might be better for consistency...

Reply:
I think just having the compiler fail is fine for now.